src/components/dashboard-widgets/dashboard-text-widget/dashboard-text-widget.component.ts

```diff
@@ -0,0 +1,79 @@
+import {ChangeDetectionStrategy, Component, ElementRef, EventEmitter, Input, Output, ViewChild} from '@angular/core';
+import {PredefinedWidgetConfig} from '../interfaces/predefined-widget.interface';
+import {TextWidgetConfig, TextWidgetLocalizedStrings} from '../interfaces/text-widget.interface';
+import {SidePanelComponent} from '../../side-panel';
+import {coerceBooleanProperty, coerceNumberProperty} from '@angular/cdk/coercion';
+
+@Component({
+    selector: 'ux-dashboard-text-widget',
+    templateUrl: './dashboard-text-widget.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DashboardTextWidgetComponent implements PredefinedWidgetConfig, TextWidgetConfig {
+    static ngAcceptInputType_fixedMode: boolean | string;
+    static ngAcceptInputType_colSpan: number | string;
+    static ngAcceptInputType_rowSpan: number | string;
+
+    private _fixedMode: boolean = false;
+    private _colSpan: number = 1;
+    private _rowSpan: number = 1;
+
+    @Input() get fixedMode(): boolean {
+        return this._fixedMode;
+    }
+    set fixedMode(fixedMode: boolean) {
+        this._fixedMode = coerceBooleanProperty(fixedMode);
+    }
+
+    @Input() get colSpan(): number {
+        return this._colSpan;
+    }
+    set colSpan(colSpan: number) {
+        this._colSpan = coerceNumberProperty(colSpan);
+    }
+
+    @Input() get rowSpan(): number {
+        return this._rowSpan;
+    }
+    set rowSpan(rowSpan: number) {
+        this._rowSpan = coerceNumberProperty(rowSpan);
+    }
+
+    @Input() id: string = '';
+    @Input() name: string = '';
+    @Input() heading: string = '';
+    @Input() localizedStrings: TextWidgetLocalizedStrings = {
+        save: 'Save',
+        cancel: 'Cancel',
+        edit: 'Edit',
+        showFullText: 'Show full text'
+    };
+
+    @Input() text: string = '';
+    @Input() editable: boolean = false;
+
+    @ViewChild('sidePanel') sidePanel: SidePanelComponent;
+    @ViewChild('textArea') textArea: ElementRef<HTMLTextAreaElement>;
+
+    @Output() textChange = new EventEmitter<string>();
+
+    open() {
+        this.textArea.nativeElement.value = this.text;
+        this.sidePanel.openPanel();
+        if (this.editable) {
+            setTimeout(() => {
+                this.textArea.nativeElement.focus();
```

Comment:
Use the `uxFocusIf` instead of this.
https://uxaspects.github.io/UXAspects/#/components/utilities#focus-if